### PR TITLE
#895 - Improve table used on the /settings/api page

### PR DESCRIPTION
**Pull request by rupin90**
Here is the behavior of the Table on different screen sizes. Finally was able to lay out the CSS. Somehow it wasn't as straight forward as I thought.
Leveraged NPM package - https://github.com/coston/react-super-responsive-table

Remaining Items:-
- [X] - border and background CSS for the table rows
- [X]  - Any feedback

https://user-images.githubusercontent.com/107437442/180309832-dcfebc78-6ffd-40fe-b609-c8729df81711.mp4

### CODE REVIEW

**Comment by jacksonh** [comment]
I think this is just starting out, right? I notice that when scaling the table down the headers don't collapse correctly:

https://www.loom.com/share/e796b6f5aae444ec9aae7b736e4d5d76

**Review by jacksonh — APPROVED** [review]

**Comment by jacksonh** [comment]
@rupin90 oh sorry i mean they dont move with the table, we do want to show them

**Comment by jacksonh** [comment]
Hey @rupin90 I noticed two things:

1. It looks like we should import TableR instead of table. `Type error: '"../../components/elements/Table"' has no exported member named 'Table'. Did you mean 'TableR'?`

2. For some reason the z-index or something is higher with the table than the modals. When I open the modal all the headers of the table appear on top of the modal.

Screen Shot 2022-07-19 at 15 31 56

**Comment by rupin90** [comment]
> Hey @rupin90 I noticed two things:
> 
> 1. It looks like we should import TableR instead of table. `Type error: '"../../components/elements/Table"' has no exported member named 'Table'. Did you mean 'TableR'?`
> 2. For some reason the z-index or something is higher with the table than the modals. When I open the modal all the headers of the table appear on top of the modal.
> 
> Screen Shot 2022-07-19 at 15 31 56

Thank you, wil look into it tonight 🙏

**Comment by rupin90** [comment]
> 

I have updated the name to `TableNew`, I can update to any other name if you have any suggestions. This is simply to avoid conflict with the library. It is working fine.

Z-index issue is fixed now. Modal didn't have any z-index so I just gave it to Modal since that would be always be the top view if `open`.

**Comment by rupin90** [comment]
Also I have tried different ways to customize the layout of the table - it has been very tricky so I have asked the community on how to customize the look and feel for the table - https://github.com/coston/react-super-responsive-table/issues/382

The `css` prop/attribute does not work on these elements. :(.

I also tried the typical way adding a className but the output is like 
``` <div class=[object object]>........ </div>``` 

Not sure if this was faced before in our codebase.

CC: @jacksonh @satindar

**Comment by jacksonh** [comment]
Hey @rupin90 this is really good. 

I just pushed a commit to show how to style these imported elements, i also renamed the import just so we could use `Table` elsewhere. 

with stitches we use this `stytled` function to make a component stylable. It also works on strings like `styled('br', { .. a bunch of css`. You can add styling in that `styled` call, or any component that you've made styled, then you can use the `css` attribute.

**Comment by rupin90** [comment]
> Hey @rupin90 this is really good.
> 
> I just pushed a commit to show how to style these imported elements, i also renamed the import just so we could use `Table` elsewhere.
> 
> with stitches we use this `stytled` function to make a component stylable. It also works on strings like `styled('br', { .. a bunch of css`. You can add styling in that `styled` call, or any component that you've made styled, then you can use the `css` attribute.

Aah nice, Thank you 🙏🏽 . Will do it just now.

File: packages/web/components/elements/Table.tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ export function TableNew(props: TableProps): JSX.Element {
           },
         }}
       >
-        <Table>
+        <ResponsiveTable>
```
Comment (rupin90) [inline packages/web/components/elements/Table.tsx]: 👏🏽  Cool!

File: packages/web/components/elements/Table.tsx [hunk 2]
```diff
@@ -106,7 +106,7 @@ export function Table(props: TableProps): JSX.Element {
           },
         }}
       >
-        <StyledTable>
+        <StyledTable css={{ /* can also style here now */ }}>
```
Comment (rupin90) [inline packages/web/components/elements/Table.tsx:109]: Thank you

**Comment by rupin90** [comment]
@jacksonh  - Thank you, this is PR ready

**Comment by jacksonh** [comment]
Hey @rupin90 one thing I notice is in small there's a black outline we don't need:


Screen Shot 2022-07-21 at 15 41 05

**Comment by rupin90** [comment]
> Hey @rupin90 one thing I notice is in small there's a black outline we don't need:
> 
> Screen Shot 2022-07-21 at 15 41 05

You mean the black border around the table? okay I can remove that.

**Comment by rupin90** [comment]
### Hey @jacksonh   - how is this?

Screen Shot 2022-07-22 at 4 25 57 PM

**Comment by jacksonh** [comment]
This looks great, thanks. I'm going to merge it now.

**Pull request merged**